Mage.Server/src/main/java/mage/server/Session.java

```diff
@@ -276,8 +276,8 @@ public String connectUserHandling(String userName, String password, String resto
         if (newUser == null) {
             User anotherUser = managerFactory.userManager().getUserByName(userName).orElse(null);
             if (anotherUser != null) {
-                boolean canDisconnectAuthDueAnotherInstance = managerFactory.configSettings().isAuthenticationActivated();
-                boolean canDisconnectAnonDueSameHost = !managerFactory.configSettings().isAuthenticationActivated()
+                boolean canDisconnectAuthDueAnotherInstance = managerFactory.configSettings().shouldCheckUsers();
+                boolean canDisconnectAnonDueSameHost = !managerFactory.configSettings().shouldCheckUsers()
```

Comment:
I don’t test it but disconnect another instance of the user is very important for xmage game engine (it can work with single user instance only). There are must be 1 user per player (registered or anon — each mode must keep single user). So game will see only 1 user to send and wait feedback.

On login logic:
- in registered mode: disconnect all other user instances;
- in anon mode: disconnect all other user instances if it’s same host (ip address).it’s important to restrict login with same name but diff IP (e.g. you can’t disconnect another logged user).

So user instance consistency must be enabled all the time. No need to setup it.

Reply:
Thank you for pointing that out!
I will do some testing on my own as well.